Frameworks/include/CoreTextInternal.h

```diff
@@ -32,38 +36,48 @@
 
 @interface _CTFrameSetter : NSObject {
 @public
-    idretaintype(_CTTypesetter) _typesetter;
+    StrongId<_CTTypesetter> _typesetter;
 }
 @end
 
 @interface _CTFrame : NSObject {
 @public
     _CTFrameSetter* _frameSetter;
     CGRect _frameRect;
+    StrongId<NSMutableArray> _lines;
+
+    // TODO::
+    // Do we need these anymore?
     CGSize _totalSize;
-    idretaintype(NSMutableArray) _lines;
     std::vector<CGPoint> _lineOrigins;
 }
 @end
 
 @interface _CTLine : NSObject {
 @public
-    NSRange _strRange;
+    _CTTypesetter* _typeSetter;
+    CFRange _strRange;
+    CGPoint _lineOrigin;
```

Comment:
It would give it for every glyph run but it is easy to calculate it for a line and we need line origins in CoreText so instead of calculating it on every access I cached it in the CTLine object

---
In reply to: [77732628](https://github.com/Microsoft/WinObjC/pull/950#discussion_r77732628) [](ancestors = 77732628)